Notion - Independent Publisher

sekharmalla opened the pull request:
Notion is an application that provides components such as notes, databases, kanban boards, wikis, calendars and reminders. These components and systems can be used individually, or in collaboration with others.

Authored-By: Chandra <43902003+sekharmalla@users.noreply.github.com>
Co-Authored-By: Harshini <65181903+harshinivarma@users.noreply.github.com>


---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [x] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [x] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [x] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [x] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

sriyen-msft commented on independent-publisher-connectors/Notion/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,675 @@
+    {
+      "propertyName": "Privacy policy",
+      "propertyValue": "https://www.notion.so/Privacy-Policy-3468d120cf614d4c9014c09f6adc9091"
+    }
```
@sekharmalla All the metadata properties are required. Can you please add the `Catagories` section as well ?

Docs - [Adding Metadata](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#step-3b-product-or-end-service-metadata)

sekharmalla replied:
Done added and pushed.

sriyen-msft commented on independent-publisher-connectors/Notion/apiProperties.json:
```diff
@@ -0,0 +1,23 @@
+        }
+      }
+    },
+    "iconBrandColor": "#F6E8EB",
```
@sekharmalla Please update it to the default iconBrandColor of all the independent publisher connectors.

`"iconBrandColor": "#da3b01"`

sekharmalla replied:
successfully updated.

sriyen-msft commented on independent-publisher-connectors/Notion/settings.json:
```diff
@@ -0,0 +1,9 @@
+{
```
@sekharmalla The settings.json file is not required, please remove it from the PR.

sekharmalla replied:
Have done it thanks.

sekharmalla commented on independent-publisher-connectors/Notion/Readme.md:
```diff
@@ -0,0 +1,97 @@
+  <a href="https://www.notion.so">
+    <svg width="87" height="30" viewBox="0 0 87 30" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M1.805 1.294l16.6-1.226c2.039-.175 2.563-.057 3.845.875l5.299 3.733c.874.642 1.165.817 1.165 1.516v20.473c0 1.283-.466 2.042-2.097 2.158L7.34 29.99c-1.224.058-1.807-.117-2.448-.934L.99 23.982c-.7-.934-.99-1.633-.99-2.45V3.334c0-1.049.466-1.924 1.805-2.04z" fill="#fff"></path><path fill-rule ...
+  </a>
+
```
Noted, i am on it.

sekharmalla commented on independent-publisher-connectors/Notion/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,675 @@
+  "consumes": [],
+  "produces": [],
+  "paths": {
+    "/databases/c9bc0f8b735f4b089c91e8a5e93df396/query": {},
```
Sure got it.

sekharmalla commented on independent-publisher-connectors/Notion/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,675 @@
+                "object": {
+                  "type": "string",
+                  "description": "object",
+                  "title": "",
```
Thanks for the right spotting. i have forgot to update it's title. i have fixed it just now.

sekharmalla commented:
@sriyen-msft  i have made all the changes and pushed to dev branch. Please let me know what are the next steps to take or if it's being reviewed that's great. 

Thanks.

sriyen-msft commented on independent-publisher-connectors/Notion/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,677 @@
+    "/databases/493fb17a-fa21-4f1b-8b46-c243894350c9/query/item": {},
+    "/databases/": {},
+    "/databases/database_id": {},
+    "/databases": {},
```
@sekharmalla You have not removed all the empty operations. Please push all the changes and have one round of check of your connector artifacts before replying back to the comment or PR.

sekharmalla replied:
Sorry for the delay. i have made the changes. As you said i have pushed all changes once and marking now as resolved. Thanks for the tip.

sriyen-msft commented on independent-publisher-connectors/Notion/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,677 @@
+    },
+    {
+      "propertyName": "Categories",
+      "propertyValue": "Productivity; knowledge management;collaboration"
```
@sekharmalla If you read the instructions, it is allowed to add a maximum of two categories. 

![image](https://user-images.githubusercontent.com/56276394/135920078-b38fa3b0-cb38-4bf2-aa48-dcdd57d5bf2d.png)

sriyen-msft commented on independent-publisher-connectors/Notion/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,677 @@
+                "object": {
+                  "type": "string",
+                  "description": "object",
+                  "title": "",
```
@sekharmalla Please remove **all** the empty operations until unless you have a specific reason to keep them.

sekharmalla commented on independent-publisher-connectors/Notion/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,668 @@
+                "type": {
+                  "type": "string",
+                  "description": "type",
+                  "title": "",
```
Sorry i left that out. Now it's fixed. Thanks for your patience.

sekharmalla commented:
Hi @sriyen-msft, Thank you for Validating, i think i have removed '_' from all the OperationID now i have fixed it. Please let me know if there are more changes. also My Timezone is GMT+4 so in case late update from me please understand.

sriyen-msft commented:
> Hi @sriyen-msft, Thank you for Validating, i think i have removed '_' from all the OperationID now i have fixed it. Please let me know if there are more changes. also My Timezone is GMT+4 so in case late update from me please understand.

@sekharmalla Please fix all the errors reported in my previous message. 

`
Connector archive validation failed:`

`Archive_The 'operationId' property value is different from its sanitized value 'RetrievesUser'. Please avoid the useage of non-alphanumeric characters to have matching values. ValuePath: /paths/~1users~1{user_id}/get/operationId`

`Archive_The 'operationId' property value is different from its sanitized value 'Retrieve-a-block'. Please avoid the useage of non-alphanumeric characters to have matching values. ValuePath: /paths/~1blocks~1{block_id}/get/operationId`

sekharmalla commented:
@sriyen-msft, Sorry for that, now everything should be fine please check, Thanks.

sekharmalla commented:
Hi @sriyen-msft, I have now fixed recommended also and validated successfully this time. Please let me know if anything else has to be done. Thank you. 

![image](https://user-images.githubusercontent.com/43902003/137034393-42947bac-66df-4a0b-b277-fa742ac76d09.png)
